eZ/Bundle/EzPublishCoreBundle/Imagine/Filter/Loader/OverlayBaseLoader.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace eZ\Bundle\EzPublishCoreBundle\Imagine\Filter\Loader;
+
+use Imagine\Filter\Basic\Fill;
+use Imagine\Image\Fill\Gradient\Horizontal;
+use Imagine\Image\Fill\Gradient\Vertical;
+use Imagine\Image\ImageInterface;
+use Imagine\Image\Point;
+use Imagine\Image\Box;
+use Imagine\Exception\InvalidArgumentException;
+
+class OverlayBaseLoader extends FilterLoaderWrapped
+{
+    public function load(ImageInterface $image, array $options = array())
+    {
+        if (!isset($options['opacity'], $options['startColor'], $options['endColor'], $options['linerClass'])) {
+            throw new InvalidArgumentException('Missing one of required options');
```

Comment:
You may want to use Symfony's OptionResolver for this. You can check the [OptionsResolverBasedQueryType](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/QueryType/OptionsResolverBasedQueryType.php) for an example.